unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMeshCombiner/AvatarMeshCombinerHelper.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Assertions;
+using Object = UnityEngine.Object;
+
+namespace DCL
+{
+    /// <summary>
+    /// AvatarMeshCombinerHelper uses the AvatarMeshCombiner utility class to combine many skinned renderers
+    /// into a single one.
+    ///
+    /// This class will recycle the same gameObject and renderer each time it is called,
+    /// and binds the AvatarMeshCombiner output to a proper well configured SkinnedMeshRenderer. 
+    /// </summary>
+    public class AvatarMeshCombinerHelper : IDisposable
```

Comment:
Out of curiosity... I was wondering why couldn't this helper be static and work with output data packages

Reply:
Yes, I decided to go for this because it needs a state. The state is the `gameObject` container. As this container is reused on successive combine calls, it has to be tracked. Also it needs to be disposed afterwards. Before creating this component, all this logic was on `AvatarRenderer` but I decided to wrap it all on this helper to be able to test it easier, use in isolated environments (like visual tests) et cétera.